db/migrations/3_claim_ticker_to_id.sql

```diff
@@ -0,0 +1,52 @@
+-- NOTE: This migration should only run after all pre 7.0 blocks have been processed
+-- Similar logic should be triggered when the 7.0 upgrade is reached
+
```

Comment:
Migration are supposed to be idempotent whenever they get executed. We never know when one decides they should run. What if someone starts the sync from 16.0.0 (which is compatible with dual version) and then bumps it in between.

Reply:
If the start with v16 there are two cases, where they update before syncing v7 upgrade and after.

If before, then the v7 block upgrade code will do the needed updates (only IDs are needed like you pointed out).

If after, then this migration will take care of it.

The `WITH spec AS` subquery is supposed to grab the latest block processed and be checked in the updates to handle this.

Reply:
Makes sense. I was thinking there was no such thing as asset IDs on 6.0 for some reason, but guess thats not the case with the calculate legacy for ticker function.